src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -464,6 +465,66 @@ private TabItem AddExtensionTab(IViewExtension viewExtension, UIElement content)
             return tab;
         }
 
+        /// <summary>
+        /// Dock the window to right side bar panel.
+        /// </summary>
+        /// <param name="window">the window which needs to be docked</param>
+        /// <param name="nodeModel">nodemodel if it is a node window</param>
+        /// <param name="hideUIElement">Any UI element to hide after docking it in the right side panel(like the toolbar).</param>
+        /// <returns></returns>
+        internal TabItem DockWindowInSideBar(Window window, NodeModel nodeModel = null, UIElement hideUIElement = null)
+        {
+            var tabItem = ExtensionTabItems.OfType<TabItem>().SingleOrDefault(tabtem => tabtem.Uid.ToString() == window.Uid);
+            if (tabItem != null)
+            {
+                tabDynamic.SelectedItem = tabItem;
+                return tabItem;
+            }
+
+            tabDynamic.DataContext = null;
+
+            if (hideUIElement != null)
+            {
+                hideUIElement.Visibility = Visibility.Collapsed;
+            }
+
+            var content = window.Content;
+
+            // creates a new tab item
+            var tab = new TabItem();
+            tab.Header = window.Title;
+            tab.Tag = window;
+            tab.Uid = window.Uid;
+            tab.HeaderTemplate = tabDynamic.FindResource("TabHeader") as DataTemplate;
+
+            // setting the extension UI to the current tab content 
+            // based on whether it is a UserControl element or window element. 
+            if (content is Window container)
+            {
+                content = container.Content as UIElement;
+                // Make sure the extension window closes with Dynamo
+                container.Owner = this;
+            }
+            tab.Content = content;
+
+            //Insert the tab at the end
+            ExtensionTabItems.Insert(ExtensionTabItems.Count, tab);
+
+            tabDynamic.DataContext = ExtensionTabItems;
+            tabDynamic.SelectedItem = tab;
+
+            Analytics.TrackEvent(Actions.Dock,
+                     Categories.ViewExtensionOperations, tab.Header.ToString());
```

Comment:
You may want to track it differently here or not track it for now

Reply:
Yeah removed it for now.